provisioning/device/tests/ProvisioningDeviceClientTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.UnitTests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class ProvisioningDeviceClientTests
+    {
+        private static readonly string s_globalDeviceEndpoint = "global.azure-devices-provisioning.net";
+        private static readonly string s_idScope = "fakeidscope";
+
+        private static readonly string s_fakeRegistrationId = "registrationId";
+        private static readonly string s_fakePrimaryKey = "dGVzdFN0cmluZbB=";
+        private static readonly string s_fakeSecondaryKey = "wGVzdFN9CmluZaA=";
+
+#pragma warning disable SYSLIB0026 // Type or member is obsolete
+        private static readonly X509Certificate2 s_cert = new();
+#pragma warning restore SYSLIB0026 // Type or member is obsolete
+        private static readonly X509Certificate2Collection s_certs = new();
+
+        [TestMethod]
+        public void ProvisioningDeviceClient_AuthProviderSymmetricKey_Works()
+        {
+            // arrange
+            var authProvider = new AuthenticationProviderSymmetricKey(s_fakeRegistrationId, s_fakePrimaryKey, s_fakeSecondaryKey);
+
+            // act
+            var provisioningDeviceClient = new ProvisioningDeviceClient(s_globalDeviceEndpoint, s_idScope, authProvider);
+
+            // assert
+            provisioningDeviceClient.Should().NotBeNull();
```

Comment:
I don't think it could ever be null. It would either throw an exception or succeed.

So, you could turn it into a function and assert it doesn't through.